doc/plugin_agent_nodeattestor_httppop.md

```diff
@@ -0,0 +1,47 @@
+# Agent plugin: NodeAttestor "httppop"
+
+*Must be used in conjunction with the server-side httppop plugin*
+
+The `httppop` plugin handshakes via http to ensure the agent is running on a valid
+dns name.
+
+The SPIFFE ID produced by the server-side `httppop` plugin is based on the dns name of the agent.
+The SPIFFE ID has the form:
+
+```xml
+spiffe://<trust_domain>/spire/agent/httppop/<hostname>
+```
+
+| Configuration     | Description                                                                                                                                      | Default   |
+|-------------------|--------------------------------------------------------------------------------------------------------------------------------------------------|-----------|
+| `hostname`        | Hostname to use for handshaking. If unset, it will be automatically detected.                                                                    |           |
+| `agentname`       | Name of this agent on the host. Useful if you have multilpe agents bound to different spire servers on the same host.                            | "default" |
+| `port`            | The port to listen on.                                                                                                                           | 80        |
```

Comment:
I think we should choose a random port here by default. Low port numbers require root which we don't always have. Defaulting to a static number can be error prone since the port might be in use.

Reply:
acme chose to standardize on port 80 as its most likely to make it across the internet unhurt. If we choose a random port, I think we're probably going to get a bunch of support requests asking why the plugin is broken? :/

But maybe I'm assuming something here. Do we think intranet usage will be the most common and spire-agent -> spire-server will be uncommon?

Reply:
> acme chose to standardize on port 80 as its most likely to make it across the internet unhurt

Well it's also the standard HTTP port for web facing services, and ACME traditionally fills requests for web facing services. This case feels different.

> But maybe I'm assuming something here. Do we think intranet usage will be the most common and spire-agent -> spire-server over the internet will be uncommon?

I do think internet traversal of this traffic is uncommon. We use mTLS there, which frequently has trouble across the internet (e.g. corporate and ISP TLS interception). My bet is that cost/benefit will outweigh use of port 80 - cons: root required, may already be in use ... pros: less likely to be filtered by a firewall. Agent/server traffic currently defaults to port 8081. If I can't or don't want to use port 80, I have to choose some other random static port number, which also feels funny

Reply:
> Well it's also the standard HTTP port for web facing services, and ACME traditionally fills requests for web facing services. This case feels different.

I have seen a fair amount of use of certbot in http mode for not webservers. But, it is much more common to usefor webservers.

> > But maybe I'm assuming something here. Do we think intranet usage will be the most common and spire-agent -> spire-server over the internet will be uncommon?
> 
> I do think internet traversal of this traffic is uncommon. We use mTLS there, which frequently has trouble across the internet (e.g. corporate and ISP TLS interception). My bet is that cost/benefit will outweigh use of port 80 - cons: root required, may already be in use ... pros: less likely to be filtered by a firewall. Agent/server traffic currently defaults to port 8081. If I can't or don't want to use port 80, I have to choose some other random static port number, which also feels funny

Where I think it may get used on the internet is for things like edge computing, where you have one spire server on the internet, and then you have a spire-agent at multiple different organizations. Likely in that scenerio, spire-server would be setup on port 443 which would make it out of all the orgs that were deploying the agent, and would need port 80 made back into each organization. So, getting N number of firewall teams to let in a particular port 80 to a host might be easier then random port at N orgs.

Again, not sure how common this will be, but somehow supporting the mode of operation for those that may need to do this kind of thing seems useful.

Reply:
I'm kind of iffy on the con list too:
> cons: root required, may already be in use

I kind of see those in some ways as extra security checks rather then cons. But I agree some may see it that way, and 
thats one of the reasons to have flags to allow_non_root_ports and allow_alternate_ports.
Maybe we should have an agent slide "use_random_port" flag too? Then it could be configured both ways.

Reply:
How about we change up the args...
On the agent, if no port is specified, pick a random one. This still allows port 80 when desired.

On the server, allow all non root and alternate ports by default but still keep them for those that want to lock down the system further?

Reply:
> On the agent, if no port is specified, pick a random one. This still allows port 80 when desired.
> On the server, allow all non root and alternate ports by default but still keep them for those that want to lock down the system further?

❤️ I like this much better